master/job.go

```diff
@@ -10,8 +11,28 @@ import (
 
 type job struct {
 	lib.Job
-	id    int
-	tasks []*lib.Task
+	id           int
+	tasks        []*lib.Task
+	runningTasks int
+	maxTasks     int
+}
+
+func (j *job) reachedMaxTasks() bool {
+	return j.runningTasks < j.maxTasks
+}
+
+func (j *job) increaseRunningTasks() {
+	if j.runningTasks >= len(j.tasks) || j.runningTasks >= j.maxTasks {
+		log.Println("Trying to run more tasks than possible or allowed.")
```

Comment:
@kc1212 @victor-li In both cases (`increaseRunningTasks` and `decreaseRunningTasks), it is possible to do `log.Fatalln` to terminate the program with exit code 1. What do you guys prefer?